### ISSUE TITLE
credentials re-written even when valid

### ISSUE THREAD

**Issue by jasonbrackman** [anson-vandoren/esp8266-captive-portal#3]
Each restart of the esp8266 that has valid credentials currently rewrites the file.  

While this _should_ be fine, it is possible to write a partial file if the write is interrupted.

I wasn't 100% sure if there was a better place, but imagine that the `write_creds()` in captive_portal.py could do the following?

```
    def write_creds(self, ssid, password):
        try:
            os.stat(self.CRED_FILE)
        except OSError:
            open(self.CRED_FILE, "wb").write(b",".join([ssid, password]))
            print("Wrote credentials to {:s}".format(self.CRED_FILE))
```

and the `try_connect_from_file()` func would need to either remove the else clause if the ssid/password is not present or an `os.remove(self.CRED_FILE) would need to be added

**Comment by anson-vandoren** [comment]
@jasonbrackman I don't have time to look too deeply at this right now, but if you want to submit a PR I'll happily review and merge.

**Comment by jasonbrackman** [comment]
I didn't go through the tutorial -- so these changes may not make sense for your needs.  
- Note that I wasn't 100% sure of ln 128 of captive_portal.py -- I changed the self.ssid to self.essid
- To test:
1. wipe the creds file from the esp8266
2. on startup the captive portal should kick in
3. Fill in valid credentials
4. It should connect to the network and do whatever is next.
5. Restart and the creds are loaded and reused, but no more rewriting.

**Comment by anson-vandoren** [comment]
@jasonbrackman thanks for the PR!

It all looks good except for the spot you questioned on Line 128. After you type in your WiFi credentials, CaptiveHttp will redirect you back to the root route (on the captive portal) while it tries to connect. Once the connection succeeds, it displays the page `connected.html`, but with the name of the network you just connected to, read from `self.ssid`. With the change you made, (`CaptivePortal.essid` instead), this page will instead show that you are connected to the ESP8266's SSID instead of the new network's.

What about giving `CaptivePortal` a `Creds` instance attribute (replacing previous `self.ssid` and `self.password`)? Otherwise you'd need to pass around your existing `Creds` object, or instantiate a new one inside `CaptivePortal.captive_portal()`

**Comment by jasonbrackman** [comment]
I have not had a lot of experience with PRs and git :).  I see that it still shows I have a conflict, but that is not the case in the latest PR.  Do I just click the resolved button?  

I ended up creating the `Creds` instance as suggested and put back the original ssid.

**Comment by anson-vandoren** [comment]
Thanks @jasonbrackman. The conflict was from a PR that @grey27 submitted the other day to fix up the credentials file never getting closed (which you also fixed, but in a different file). I resolved it and merged your PR. Thanks again

**Comment by anson-vandoren** [comment]
Fixed in #4
